src/components/Class/index.js

```diff
@@ -939,59 +987,62 @@ function Class({ classToEdit, indicator }) {
-                      changeHandler(e, setFormFieldsState, formFieldsState)
-                    }
-                    value={formFieldsState[OCCURRENCE]}
-                    placeholder="How many recurring classes"
-                    disabled={isEditMode && !indicator ? true : false}
-                    required={
-                      formFieldsState[TYPE] === "cohort" &&
-                      formFieldsState[UNTIL] === ""
-                        ? true
-                        : false
-                    }
-                    max={48}
-                  />
+                  {pathwayName !== "python" && (
+                    <>
+                      <label htmlFor={UNTIL} className="label-field">
+                        Until
+                        <span className="optional-field">
+                          (either until or occurrence is required)
+                        </span>
+                      </label>
+                      <input
+                        className="input-field input-field--short"
+                        type="date"
+                        data-date-format="YYYY MM DD"
+                        name={UNTIL}
+                        id={UNTIL}
+                        onChange={(e) =>
+                          changeHandler(e, setFormFieldsState, formFieldsState)
+                        }
+                        value={formFieldsState[UNTIL]}
+                        placeholder="Until when recurring classes"
+                        disabled={isEditMode && !indicator ? true : false}
+                        required={
+                          formFieldsState[TYPE] === "cohort" &&
+                          formFieldsState[OCCURRENCE] === ""
+                            ? true
+                            : false
+                        }
+                      />
+                      <label htmlFor={OCCURRENCE} className="label-field">
+                        Occurrence
+                        <span className="optional-field">
+                          (either occurrence or until is required)
+                        </span>
+                      </label>
+                      <input
+                        className="input-field"
+                        type="number"
+                        name={OCCURRENCE}
+                        id={OCCURRENCE}
+                        onChange={(e) =>
+                          changeHandler(e, setFormFieldsState, formFieldsState)
+                        }
+                        value={formFieldsState[OCCURRENCE]}
+                        placeholder="How many recurring classes"
+                        disabled={isEditMode && !indicator ? true : false}
+                        required={
+                          formFieldsState[TYPE] === "cohort" &&
+                          formFieldsState[UNTIL] === ""
+                            ? true
+                            : false
+                        }
```

Comment:
Jab aapke paas ek Boolean condition that matches expression ka value, expression zaroori nahi hai. (Hope my mix of Hindi was correct; main seekhane kee koshish kar raha hoon.)
```suggestion
                        disabled={isEditMode && !indicator}
                        required={
                          formFieldsState[TYPE] === "cohort" &&
                          formFieldsState[UNTIL] === ""
                        }
```